### ISSUE TITLE
Update the node executable version shipped with installers

### ISSUE THREAD

**Issue by tet3** [forcedotcom/cli#281]
**Is your feature request related to a problem? Please describe.**
The [downloadable installers](https://developer.salesforce.com/tools/sfdxcli), at least for Windows and macOS, ship with a nodejs executable, currently v10.15.3. This executable is the running version of Node for sfdx commands, even if a different version of node is installed globally. Some sfdx extensions, like [dependencies-cli](https://github.com/forcedotcom/dependencies-cli), have dependencies on newer versions of Node, and [cannot be installed if using the installer version of sfdx](https://github.com/forcedotcom/dependencies-cli/issues/4). 

At least on Windows, copying a newer version (v12.14.1) of node.exe over top of that shipped with with the installer, causes no issues with sfdx functionality. And installing sfdx with npm in a node v12.14.1 environment also works well, so it seems like it should be straightforward to replace the executable shipped with the installers. 

**What are you trying to do**
Install sfdx extensions that require later versions of Node than that shipped with the sfdx installers.

**Describe the solution you'd like**
Ship the most current Node LTS executable with the downloadable installers. 

**Describe alternatives you've considered**
Uninstalling the installer-based sfdx installation and installing via npm resolves the compatibility issue for extensions that rely on node >10.15.3.

**Comment by kacrouse** [comment]
For the solution, I'd prefer it to check if node is installed locally and use that if it is (similar to how the npm version functions), otherwise use the version it ships with.


**For those on macOS**
The node binary sfdx ships with can be tricky to find, it's stored in `~/.local/share/sfdx/client/current/bin`

Or if you just want to make it use your local node binary, you can run the following command:
```bash
rm ~/.local/share/sfdx/client/current/bin/node && ln -s $(which node) ~/.local/share/sfdx/client/current/bin/node
```
You will probably have to rerun this every time sfdx updates though 😢

**Comment by amphro** [comment]
Closing this out as the CLI is currently shipped with 12.18.3. We will probably update it to v14 soon. 

I'm hesitant to use the currently installed version of node. If the user really wants to use their version of node, they can install the CLI with a package manager (npm, yarn, etc). If that is something really desirable, create a new issue describing that feature request to see if there is enough community demand for it.
